### PR TITLE
feat: add a GraphRAG query length config

### PR DESCRIPTION
**Pull request by imbajin**
This commit renames the configuration parameter for GraphRAG's maximum query length, as per your feedback, to make it shorter and more concise.

Changes:
- Renamed `graph_rag_max_query_length` to `rag_query_max_length` in `LLMConfig`.
- Corresponding environment variable changed from `GRAPH_RAG_MAX_QUERY_LENGTH` to `RAG_QUERY_MAX_LENGTH`.
- Updated `GraphRAGQuery` and its unit tests to use the new names.
- Updated `hugegraph-llm/README.md` to reflect the new environment variable name.

## Summary by CodeRabbit

- **新功能**
  - 增加了对RAG相关查询输入长度的限制，默认最大字符数为50，超出将被拒绝处理。
  - 支持通过环境变量自定义最大查询长度，可在`.env`文件或Web界面进行配置。
- **文档**
  - README文档新增了环境变量配置说明，详细介绍了RAG查询长度限制的设置方法和相关变量。
- **Bug修复**
  - 对RAG API及相关演示界面增加了输入长度校验，防止过长查询导致异常。
- **测试**
  - 新增了针对RAG查询长度校验的单元测试，覆盖API接口和演示功能的正常与异常场景。

### CODE REVIEW

**Comment by imbajin** [comment]
@codecov-ai-reviewer review

File: hugegraph-llm/src/hugegraph_llm/config/llm_config.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ class LLMConfig(BaseConfig):
     text2gql_llm_type: Literal["openai", "litellm", "ollama/local", "qianfan_wenxin"] = "openai"
     embedding_type: Optional[Literal["openai", "litellm", "ollama/local", "qianfan_wenxin"]] = "openai"
     reranker_type: Optional[Literal["cohere", "siliconflow"]] = None
+    rag_query_max_length: int = os.environ.get("RAG_QUERY_MAX_LENGTH", 50)
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/config/llm_config.py:33]: 🚨 [Critical] Missing proper type conversion for environment variable

The environment variable is retrieved without proper conversion to int. When `RAG_QUERY_MAX_LENGTH` is set in the environment, it will be a string, but the default value is an integer (50), causing type inconsistency.

```suggestion
rag_query_max_length: int = int(os.environ.get("RAG_QUERY_MAX_LENGTH", "50"))
```

This ensures consistent integer type regardless of whether the value comes from environment or default.

File: hugegraph-llm/src/hugegraph_llm/api/models/rag_requests.py [hunk 2]
```diff
@@ -62,6 +62,14 @@ class RAGRequest(BaseModel):
         description="Prompt for the Text2Gremlin query.",
     )
 
+    @validator('query')
+    def check_query_length(cls, value):
+        llm_config = LLMConfig()
+        max_len = int(llm_config.rag_query_max_length)
+        if len(value) > max_len:
+            raise ValueError(f"Query exceeds maximum allowed length of {max_len} characters.")
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/api/models/rag_requests.py:70]: ⚠️ [Medium] Duplicated validation logic between RAGRequest and GraphRAGRequest

Both classes have identical `check_query_length` validators. Consider extracting to a base class or mixin to reduce duplication and ensure consistency:

```python
class QueryValidationMixin:
    @validator('query')
    def check_query_length(cls, value):
        llm_config = LLMConfig()
        max_len = int(llm_config.rag_query_max_length)
        if len(value) > max_len:
            raise ValueError(f"Query exceeds maximum allowed length of {max_len} characters.")
        return value

class RAGRequest(BaseModel, QueryValidationMixin):
    # ... existing fields

class GraphRAGRequest(BaseModel, QueryValidationMixin):
    # ... existing fields
```

This would make future maintenance easier and ensure both validators stay in sync.

File: hugegraph-llm/src/hugegraph_llm/api/models/rag_requests.py [hunk 3]
```diff
@@ -62,6 +62,14 @@ class RAGRequest(BaseModel):
         description="Prompt for the Text2Gremlin query.",
     )
 
+    @validator('query')
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/api/models/rag_requests.py:65]: ⚠️ **[Medium]** Duplicated validation logic violates DRY principle

The same query length validation is duplicated here and in rag_requests.py lines 99-105. Consider extracting to a shared validation utility function to improve maintainability.

Suggested approach: Create a validate_query_length() utility that can be reused across all validation points.

File: hugegraph-llm/src/tests/api/test_rag_api.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+            # We will construct the request, and the endpoint call will internally trigger validation
+            # if the model is instantiated there, or FastAPI handles it if passed as type hint.
+            # For this test to be accurate to FastAPI behavior for request body validation:
+            # We should not expect to catch HTTPException directly from RAGRequest instantiation here.
+            # Instead, the endpoint call should be the one raising it due to FastAPI's processing.
+            # The current test structure where `rag_answer_api_endpoint(req)` is called is correct
+            # if we assume FastAPI passes a validated model or the model instantiation happens inside.
+            # Given Pydantic validator raises ValueError, FastAPI converts this to HTTP 422.
+
+            # Simulate calling the endpoint which would trigger Pydantic validation via FastAPI
```
Comment (imbajin) [inline hugegraph-llm/src/tests/api/test_rag_api.py:76]: ⚠️ **[Medium]** Test doesn't properly validate FastAPI behavior

This test manually creates HTTPException to simulate FastAPI's validation, but doesn't actually test the real request flow. Should use FastAPI's TestClient for proper integration testing to ensure validation works correctly in production.

Example:
```python
from fastapi.testclient import TestClient
client = TestClient(app)
response = client.post('/rag', json={'query': long_query})
assert response.status_code == 422
```

**Comment by imbajin** [comment]
## Code Review Summary

**Overall Assessment**: Changes Required

### Key Findings
- 🚨 **Critical**: 1 issue (type conversion runtime error)
- ⚠️ **Medium**: 4 issues (code duplication, inconsistent errors, test structure)  
- 🧹 **Minor**: 2 issues (magic numbers, edge cases)

### Critical Issue
**llm_config.py line 33**: os.environ.get returns string but assigned to int field. Will cause TypeError. Must add int() conversion.

### Main Improvements Needed
1. Fix type conversion bug (blocker)
2. Extract duplicated validation to shared utility
3. Make error handling consistent
4. Use TestClient for proper test coverage
5. Define constants for magic numbers

See line-level comments for specific fixes.
